src/worker/tasks/sendTransactionWorker.ts

```diff
@@ -81,11 +81,48 @@ const handler: Processor<any, void, string> = async (job: Job<string>) => {
   // null = the transaction attemped to resend but was not needed. Ignore.
   // A thrown exception indicates a retry-able error occurred (e.g. RPC outage).
   let resultTransaction: SentTransaction | ErroredTransaction | null;
-  if (transaction.status === "queued") {
+
+  let populatedTransaction:
+    | Awaited<ReturnType<typeof toSerializableTransaction>>
+    | undefined;
+
+  const chain = await getChain(transaction.chainId);
+  // Populate the transaction to resolve gas values.
+  // This call throws if the execution would be reverted.
+  // The nonce is _not_ set yet.
+  try {
+    populatedTransaction = await toSerializableTransaction({
+      from: getChecksumAddress(transaction.from),
+      transaction: {
+        client: thirdwebClient,
+        chain,
+        ...transaction,
+        // Stub the nonce because it will be overridden later.
+        nonce: 1,
+      },
+    });
+  } catch (e: unknown) {
+    // If the transaction will revert, error.message contains the human-readable error.
+    const errorMessage = (e as Error)?.message ?? `${e}`;
+    job.log(`Failed to estimate gas: ${errorMessage}`);
+    resultTransaction = {
+      ...transaction,
+      status: "errored",
+      errorMessage,
+    };
+  }
+
+  // we know that if status is queued, populatedTransaction is defined
+  // typescript has a hard time understanding, but we can safely check that populatedTransaction is defined
+  if (transaction.status === "queued" && populatedTransaction) {
     if (transaction.isUserOp) {
       resultTransaction = await _sendUserOp(job, transaction);
```

Comment:
You need to pass the populated tx here though, you want to make sure you use the 'data' from the populated tx when you prep the execute call